File: go-version/leetcode/LC2239/find_closest_number.go
package lc2239

import "math"

func findClosestNumber(nums []int) int {
	ans := math.MaxInt
	for _, num := range nums {
		if abs(num) < abs(ans) {
			ans = num
		} else if abs(num) == abs(ans) {
			ans = max(ans, num)
		}
	}
	return ans
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

File: python-version/LC2239/find_closest_number.py
import math


class Solution:
    def findClosestNumber(self, nums: list[int]) -> int:
        ans = math.inf
        for x in nums:
            if math.fabs(x) < math.fabs(ans):
                ans = x
            elif math.fabs(x) == math.fabs(ans):
                ans = max(ans, x)
        return ans
    